both/methods.js

```diff
@@ -153,16 +153,25 @@ Meteor.methods({
 
     Meteor.call('refreshRoundCodes', competitionId, eventCode);
   },
-  addNonEventRound: function(competitionId, round) {
+  addScheduleEvent: function(competitionId, eventData, roundId) {
     check(competitionId, String);
     throwIfCannotManageCompetition(this.userId, competitionId);
-    return Rounds.insert({
+
+    var round = Rounds.findOne(roundId);
+
+    return ScheduleEvents.insert({
       competitionId: competitionId,
-      title: round.title,
-      startMinutes: round.startMinutes,
-      durationMinutes: round.durationMinutes,
+      roundId: (round ? round._id : null),
+      title: (round ? round.displayTitle() : eventData.title),
```

Comment:
Unfortunately, I fear that this can get out of sync. The `displayTitle` for a round changes as the round becomes combined/uncombined, or future rounds are added. For example if you have 1 round of 3x3, then that is "3x3 Final Round". However, after adding a second round of 3x3, that 1st round is now called "3x3 First round", rather than "3x3 Final Round". This is done by the `refreshRoundCodes` method [here](https://github.com/cubing/ccm/blob/4ac23a95eba1370a76bce7308c1ec1ab3abbe1bd/both/methods.js#L232). One fix might be to have `refreshRoundCodes` also write to the title field for `ScheduleEvents`. Another fix might be to not set the title for ScheduleEvents that have a `roundId`, and instead go to the corresponding Round if you want a pretty string.